pkg/storage/etcd/etcdcommon/config.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package etcdcommon
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"strings"
+
+	etcd "github.com/coreos/etcd/client"
+	"github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/pkg/transport"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/storage"
+	etcd2 "k8s.io/kubernetes/pkg/storage/etcd"
+	"k8s.io/kubernetes/pkg/storage/etcd3"
+	utilnet "k8s.io/kubernetes/pkg/util/net"
+)
+
+type ETCDStorageConfig struct {
+	Config ETCDConfig
```

Comment:
`ETCD *ETCDConfig`

That way people can add e.g. `Foo *FooBackend` and the Construct() method can do the right thing.